lib/jobs/dell-racadm-tool-job.js

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = racadmToolJobFactory;
+di.annotate(racadmToolJobFactory, new di.Provide('Job.Dell.RacadmTool'));
+di.annotate(racadmToolJobFactory, new di.Inject(
+    'Job.Base',
+    'JobUtils.RacadmTool',
+    'Logger',
+    'Util',
+    'Assert',
+    'Promise',
+    '_',
+    'Services.Encryption',
+    'Services.Lookup',
+    'Constants',
+    'Services.Waterline'
+));
+
+function racadmToolJobFactory(
+    BaseJob,
+    racadmTool,
+    Logger,
+    util,
+    assert,
+    Promise,
+    _,
+    encryption,
+    lookup,
+    Constants,
+    waterline
+) {
+    var logger = Logger.initialize(racadmToolJobFactory);
+    /**
+     *
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function racadmToolJob(options, context, taskId) {
+        racadmToolJob.super_.call(this, logger, options, context, taskId);
+
+        this.nodeId = this.context.target;
+        this.action = options.action;
+        this.cifsInfo = {user: options.server_username, password: options.server_password ,
+            filePath: options.server_filePath};
+    }
+    util.inherits(racadmToolJob, BaseJob);
+
+    /**
+     * @memberOf racadmToolJob
+     */
+    racadmToolJob.prototype._run = function() {
+        var self = this;
+        waterline.nodes.findByIdentifier(self.nodeId)
+            .then(function (node) {
+                assert.ok(node, 'No node for dell racadm set bios');
+                var obmSetting = _.find(node.obmSettings, { service: 'ipmi-obm-service' });
+                assert.ok(obmSetting, 'No ipmi obmSetting for dell racadm set bios');
+                return obmSetting.config;
+            })
+            .then(self.revealSecrets)
+            .then(self.lookupHost)
+            .then(function(obmSetting) {
+                if (self.action === 'SetBIOS'){
+                    return racadmTool.setBiosConfig(obmSetting.host, obmSetting.user,
+                        obmSetting.password,
+                        self.cifsInfo);
+                } else if (self.action === 'UpdateFirmware'){
+                    return racadmTool.updateFirmware(obmSetting.host, obmSetting.user,
+                        obmSetting.password,
+                        self.cifsInfo);
+                }else if (self.action === 'GetBIOS'){
+                    return racadmTool.getBiosConfig(obmSetting.host, obmSetting.user,
+                        obmSetting.password,
+                        self.cifsInfo);}else{
+                    throw new Error('The action specified is not supported!.');
```

Comment:
This is an example of obm-control about how to handle such `action` field: https://github.com/RackHD/on-tasks/blob/master/lib/jobs/obm-control.js#L86, it can be very simple and flexible. So please change it to that style.